src/deposit-address.ts

```diff
@@ -0,0 +1,282 @@
+// Bitcoin Segwit Address Generator
+// TypeScript implementation based on Go code
+
+import bs58 from "bs58";
+import * as crypto from "crypto";
+import {
+  getBlockchainConfig,
+  Address,
+  Ecosystem,
+  LChainId,
+  SupportedBlockchains,
+} from "./chain-id";
+import { AddressesResponse, fetchAddressMetadata } from "./api";
+import { computeAuxData } from "./aux-data";
+import {
+  sha256,
+  Networks,
+  NetworkParams,
+  BitcoinAddressError,
+} from "./bitcoin";
+import { Tweaker } from "./tweaker";
+
+const DEPOSIT_ADDR_TAG = "LombardDepositAddr";
+const DEPRECATED_CHAIN_TAG = 0;
+
+// Root deposit public keys
+export const MAINNET_PUBLIC_KEY = Buffer.from(
+  "033dcf7a68429b23a0396ca61c1ab243ccbbcc629ff04c59394458d6db5dd2bb15",
+  "hex",
+);
+export const SIGNET_PUBLIC_KEY = Buffer.from(
```

Comment:
nit, key is used for our testnet 
```suggestion
export const TESTNET_PUBLIC_KEY = Buffer.from(
```

```suggestion
export const GASTALD_PUBLIC_KEY = Buffer.from(
```